eos/rare-b-decays/charm-loops.cc

```diff
@@ -1226,4 +1239,147 @@ namespace eos
 
         return 1.0 / (1.0 - s_hat) * (2.0 * at1 * (at1 - at2) + log1 * log1 - Li_1 - Li_2 + Li_3 + Li_4);
     }
+
+    complex<double>
+    CharmLoops::F17_massive_Qsb(const double & s)
+    {
+        if (s < 0.0 or s > 15.0)
+        {
+            throw InternalError("CharmLoop::F17_massive_Qsb used outside its domain of validity, s = " + stringify(s));
+        }
+
+        return CharmSpline("F17_massive_Qsb", s);
+    }
+    complex<double>
+    CharmLoops::F19_massive_Qsb(const double & s)
+    {
+        if (s < 0.0 or s > 15.0)
+        {
+            throw InternalError("CharmLoop::F19_massive_Qsb used outside its domain of validity, s = " + stringify(s));
+        }
+
+        return CharmSpline("F19_massive_Qsb", s);
+    }
+    complex<double>
+    CharmLoops::F27_massive_Qsb(const double & s)
+    {
+        if (s < 0.0 or s > 15.0)
+        {
+            throw InternalError("CharmLoop::F27_massive_Qsb used outside its domain of validity, s = " + stringify(s));
+        }
+
+        return CharmSpline("F27_massive_Qsb", s);
+    }
+    complex<double>
+    CharmLoops::F29_massive_Qsb(const double & s)
+    {
+        if (s < 0.0 or s > 15.0)
+        {
+            throw InternalError("CharmLoop::F29_massive_Qsb used outside its domain of validity, s = " + stringify(s));
+        }
+
+        return CharmSpline("F29_massive_Qsb", s);
+    }
+
+    complex<double>
+    CharmLoops::CharmSpline(const char* contribution, const double & s)
```

Comment:
Ok, this is don't like at all. ``strcomp`` in a hotpath? Please don't.

Can you please move the spline data into a dedicated class, have a ``static const`` object of that struct per function?
We can further discuss this if you like.

Reply:
@dvandyk Please have a look and tell me it this is any better :)